pkg/scheduler/ugm/manager.go

```diff
@@ -314,15 +432,278 @@ func (m *Manager) isGroupRemovable(gt *GroupTracker) bool {
 	return false
 }
 
+func (m *Manager) UpdateConfig(config configs.QueueConfig, queuePath string) error {
+	m.Lock()
+	defer m.Unlock()
+
+	// clear the local limit config maps before processing the limit config
+	m.userLimitsConfig = make(map[string]map[string]map[string]interface{})
+	m.groupLimitsConfig = make(map[string]map[string]map[string]interface{})
+	return m.internalProcessConfig(config, queuePath)
+}
+
+func (m *Manager) internalProcessConfig(cur configs.QueueConfig, queuePath string) error {
```

Comment:
This method is excessively long. Can't you split it up to reasonable parts & name them properly?